### PR TITLE
afpacket: add a simple unit test

### PR DESCRIPTION
**Pull request by moosingin3space**

### CODE REVIEW

File: afpacket/src/sockets.rs [hunk 1]
```diff
@@ -139,6 +142,24 @@ impl Socket {
 }
 
 impl BoundSocket {
+    /// Puts this socket's NIC into promiscusous mode. Useful for recieving all packets on an
+    /// interface, including those not addressed to the device.
+    pub fn set_promisc(&mut self) -> io::Result<()> {
```
Comment (ddowl) [inline afpacket/src/sockets.rs]: `promisc` almost looked like `promise` to me. Maybe we can expand the function name to `set_promiscuous` for clarity?

Comment (ddowl) [reply afpacket/src/sockets.rs]: Should we have an option to toggle promiscuous mode on and off?

File: afpacket/tests/afpacket_tests.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    };
+    let eth_hdr = wire::EthernetRepr {
+        src_addr: wire::EthernetAddress::BROADCAST,
+        dst_addr: wire::EthernetAddress::BROADCAST,
+        ethertype: wire::EthernetProtocol::Ipv4,
+    };
+    let buf_size =
+        wire::EthernetFrame::<&[u8]>::buffer_len(ipv4_hdr.buffer_len() + udp_hdr.buffer_len());
+    let mut out_buffer = vec![0; buf_size];
+    {
```
Comment (ddowl) [inline afpacket/tests/afpacket_tests.rs]: What's with the extra block scope?

**Comment by ddowl** [comment]
@moosingin3space Neat! The loopback test looks sound to me, but it might be nice to hear about what went wrong with `af_packet` at the next meetup. Other than the clippy format and the nits, it LGTM.

**Review by ddowl — DISMISSED** [review]

File: afpacket/tests/afpacket_tests.rs (hunk 2 above)
Comment (moosingin3space) [reply afpacket/tests/afpacket_tests.rs]: Forgot that's now legal with NLL. 
(My concern was with the mutable borrow followed by the immutable borrow down below.)

File: afpacket/src/sockets.rs (hunk 1 above)
Comment (moosingin3space) [reply afpacket/src/sockets.rs]: Resolved the name.

**Review by ddowl — APPROVED** [review]
LGTMMD

**Pull request merged**